Doc update (#171)
ufs-community/ufs-mrweather-app: docs/UsersGuide/source/inputs_outputs.rst

The :term:`UFS` MR Weather App requires numerous input files. :term:`CIME` can copy/link to input files,
run the end-to-end system and write output files to disk. Depending on the dates and format
(`GRIB2 <https://www.nco.ncep.noaa.gov/pmb/docs/grib2/>`_,
`NEMSIO <https://github.com/NOAA-EMC/NCEPLIBS-nemsio/wiki/Home-NEMSIO>`_, or 

> `NETCDF <https://www.unidata.ucar.edu/software/netcdf/>`_)

Author: updated.

requested, input files can be automatically retrieved by CIME (:term:`GRIB2`) or must be staged by

> the user (:term:`NEMSIO` or :term:`NETCDF`).

Author: Updated.

The UFS MR Weather App currently only supports the use of Global Forecast System
(GFS) data as raw initial conditions (that is, MRF, AVN, ERA5 etc. are not supported).

> The GFS data can be provided in three formats: :term:`NEMSIO`, :term:`NETCDF`, or :term:`GRIB2`. Both types of files can be obtained

Author: updated.

from the `NCEI website <https://www.ncdc.noaa.gov/data-access/model-data/model-datasets/global-forcast-system-gfs>`_.

- **NEMSIO**

  These files cover the entire globe down to a horizontal resolution of 13 km and
  can be found at `<https://nomads.ncep.noaa.gov/pub/data/nccf/com/gfs/prod/>`_.  
  

> - **NETCDF**

Author: updated

  These files cover the entire globe down to a horizontal resolution of 13 km and
  can be found at the FTP data repository `<https://ftp.emc.ncep.noaa.gov/EIB/UFS/>`_.  
     
- **GRIB2**

  These files cover the entire globe and resolutions of 0.5, or 1.0 degree are supported.

  - 0.5 deg files are available at `<https://www.ncei.noaa.gov/thredds/catalog/model-gfs-g4-anl-files-old/catalog.html>`_

>   - 1.0 deg files can be requested from `<https://www.ncdc.noaa.gov/has/HAS.FileAppRouter?datasetname=GFS3&subqueryby=STATION&applname=&outdest=FILE>`_

Reviewer: ```suggestion
  - 1.0 deg files can be requested from `<https://www.ncei.noaa.gov/thredds/catalog/model-gfs-g3-anl-files-old/catalog.html>`_
```
Isn't is best to point the user to the THEDDS data server for all types of GRIB2 files?

Author: Changed, thanks!

  These files cover the entire globe and resolutions of 0.5, or 1.0 degree are supported.

  - 0.5 deg files are available at `<https://www.ncei.noaa.gov/thredds/catalog/model-gfs-g4-anl-files-old/catalog.html>`_

>   - 1.0 deg files can be requested from `<https://www.ncei.noaa.gov/hthredds/catalog/model-gfs-g3-anl-files-old/catalog.html`_

Reviewer: ```suggestion
  - 1.0 deg files can be requested from `<https://www.ncei.noaa.gov/thredds/catalog/model-gfs-g3-anl-files-old/catalog.html`_
```

Author: Changed, thanks!

- **NEMSIO**

  - Two-dimensional surface variables ``gfs.tHHz.sfcanl.nemsio``
  - Three-dimensional atmosphere state ``gfs.tHHz.atmanl.nemsio`` 

> - **NETCDF**

Author: updated

  - Two-dimensional surface variables ``gfs.tHHz.sfcanl.nc``
  - Three-dimensional atmosphere state ``gfs.tHHz.atmanl.nc`` 
 
- **GRIB2**

>   - Surface variables and atmosphere state ``gfsanl_4_YYYYMMDD_HH00_000.grb2``

Reviewer: ```suggestion
  - Surface variables and atmosphere state in 0.5 deg ``gfsanl_4_YYYYMMDD_HH00_000.grb2``
```

Author: Changed, thanks!

         chmod 755 get.sh
         ./get.sh 20191224 12

>      For downloading grib2 files, the same code ``get.sh`` can be used except replacing wget part by the following line: 

Author: updated

     .. code-block:: console

         wget -c https://www.ncei.noaa.gov/thredds/catalog/model-gfs-g4-anl-files/$yyyymmdd/gfs_4_${yyyymmdd)_${hh}00_000.grb2

>      If the file is a gfs3 file not a gfs4 file, the user must link the new file to the old file name. For example, 

Reviewer: ```suggestion
     For downloading files in GRIB2 format with 1.0 degree grid spacing, the same code ``get.sh`` can be used except the wget command should be replaced with the following line: 
     
     .. code-block:: console

         wget -c https://www.ncei.noaa.gov/thredds/catalog/model-gfs-g3-anl-files/$yyyymmdd/gfs_3_${yyyymmdd)_${hh}00_000.grb2
     
     If the file has 1.0 degree resolution (gfs3 file), the user must link the new file to the name expected by the App. For example, 
```
It is possible that this linking bug will be fixed before the release. If so, we'll update this text later.

Author: Changed, thanks!

     .. code-block:: console

         ln -s gfs_3_20190829_0000_000.grb2 gfs_4_20190829_0000_000.grb2

>      For downloading NETCDF files, the wget parts in ``get.sh`` need to be changed to:

Author: updated

     .. code-block:: console

         wget -c https://ftp.emc.ncep.noaa.gov/EIB/UFS/inputdata/$yyyymm/gfs.$yyyymmdd/$hh/gfs.t${hh}z.atmf000.nc
         wget -c https://ftp.emc.ncep.noaa.gov/EIB/UFS/inputdata/$yyyymm/gfs.$yyyymmdd/$hh/gfs.t${hh}z.sfcf000.nc

>      Currently, only a few sample NETCDF files are available for testing at the FTP data repository.

Reviewer: ```suggestion
     Currently, only two sample netCDF files are available for testing at the FTP data repository.
```

Author: Changed.

>      Currently, only a few sample NETCDF files are available for testing at the FTP data repository.

Author: Updated.

a given initialization hour and can have files for multiple initialization hours
(00, 06, 12, and 18 UTC).

> If a directory has both GRIB2 and NEMSIO (or NETCDF) files for the same initialization date and time,

Author: updated.

CIME will use the GRIB2 files. If the user wants to change this behavior so CIME uses the

> NEMSIO or NETCDF files, the user should edit file ``user_nl_ufsatm``

Author: Updated.

and add

.. code-block:: console

    input_type = "gaussian" for NEMSIO

>     input_type = "gaussian_netcdf" for NETCDF

Author: Changed.
